draco/renderer/altair/altair_renderer.py

```diff
@@ -0,0 +1,532 @@
+
+    @staticmethod
+    def __get_encoding_channel_of_field(
+        views: list[View], field_name: FieldName
+    ) -> EncodingChannel | None:
+        """
+        Returns the mark encoding channel for the field with the given name.
+
+        :param views: All the views in the spec to search for.
+        :param field_name: The name of the field to search for.
+        :return: The encoding channel of the field, or `None` if
+                 the field is not found.
+        """
+        for v in views:
+            for m in v.mark:
+                for e in m.encoding:
+                    if e.field == field_name:
+                        return e.channel
+
+        return None
+
+    @staticmethod
+    def __get_scales_of_spec(spec: SpecificationDict) -> list[Scale]:
+        """
+        Returns all the scales in the spec including the top-level
+        shared scale and the view-specific scales.
+
+        :param spec: The spec to search through for scales.
+        :return: A list of all the scales in the spec.
+        """
+        scales: list[Scale] = []
+
+        # Shared scales
+        for s in spec.scale or []:
+            scales.append(s)
+
+        # View-specific scales
+        for v in spec.view:
+            for s in v.scale or []:
+                scales.append(s)
+
+        return scales
+
+    @staticmethod
+    def __get_field_by_name(fields: list[Field], field_name: FieldName) -> Field:
+        """
+        Returns the field with the given name.
+
+        :param fields: The list of fields to search through.
+        :param field_name: The name of the field to search for.
+        :return: The field with the given name.
+        :raises ValueError: If the field is not found.
+        """
+        for f in fields:
+            if f.name == field_name:
+                return f
+
+        raise ValueError(f"Field {field_name} not found")
+
+    @staticmethod
+    def __get_field_type(spec: SpecificationDict, field_name: FieldName) -> str:
+        """
+        Returns the type of the field with the given name.
+        Needed to map from Draco-spec data types to Vega-Lite data types.
+        Also takes the scale's type into account which is associated with the field.
+
+        :param spec: the whole specification
+        :param field_name: name of the field to look up
+        :return: the type of the field
+        """
+        cls = AltairRenderer
+        __DEFAULT_KEY__ = "default"
+        # Multi-criteria lookup to determine the type of the field
+        # based on the scale type (if any) AND the raw data type.
+        renames = {
+            "linear": {
+                "number": "quantitative",
+                "datetime": "temporal",
+            },
+            "log": {
+                "number": "quantitative",
+                "datetime": "quantitative",
+            },
+            "ordinal": {
+                "number": "ordinal",
+                # Accounting for colors
```

Comment:
What does this mean? I think we can still use ordinal for now.

Reply:
Thanks for pointing this out. As soon as I adjusted this to `ordinal` I discovered a bug after seeing that the unit tests are failing 🙃

It is fixed now.